fiducial_follow/nodes/follow.py

```diff
@@ -157,26 +253,208 @@ def newTf(self, msg):
+                cmdType      = newCommand[0]
+                actOnDone    = newCommand[1]
+                cmdComment   = newCommand[2]
+                strParam1    = newCommand[3]
+                strParam2    = newCommand[4]
+                numParam1    = newCommand[5]
+                numParam2    = newCommand[6]
+
+                print "Accept New Command: %s actionOnDone %s str1 %s num1 %lf comment %s\n" % \
+                  (cmdType, actOnDone, strParam1, numParam1, cmdComment)
+
+                # default is when done fetch another command
+                self.currentCommand  = cmdType
+                self.cmdStatus       = self.cmdStatusInProgress
+                self.cmd_comment     = cmdComment
+
+                # set a default action on done to get next command UNLESS action was specified 
+                if len(actOnDone) > 2:
+                  self.actOnDone     = actOnDone
+                else:
+                  self.actOnDone     = self.actOnDoneDoNextCommand
+
+                if cmdType == self.cmdFollowFiducial:
+                    # set new fiducial to follow with required end action once we have tracked it
+                    self.target_fiducial = strParam1
+                    self.publishStatus1Str(cmdType, "SetNewFiducial", self.target_fiducial)
+
+                elif cmdType == self.cmdDriveForward:
+                    # Drive straight the specified distance. This will clear tracking till next command
+                    # numParam1 is speed in meters per second set in drive_rate in meters per sec
+                		    # Enhancement in future may be to use numParam2 as drive speed in meters per sec
+                    self.target_fiducial = self.null_fiducial
+                    
+                    # setup for cmd_vel publishing after the parsing is done
+                    self.moveTimeLeft    = numParam1
+                    driveRate            = self.drive_rate
+                    self.publishStatus1StrAndValue(cmdType, "Driving", "Drive time in seconds of ", numParam1)
+
+                elif cmdType == self.cmdDriveReverse:
+                    # Drive in reverse the specified distance. This will clear tracking till next command
+                    # numParam1 is speed in meters per second set in drive_rate in meters per sec
+                    # Enhancement in future may be to use numParam2 as drive speed in meters per sec
+                    self.target_fiducial = self.null_fiducial
+                    
+                    # setup for cmd_vel publishing after the parsing is done
+                    self.moveTimeLeft    = numParam1
+                    driveRate            = self.drive_rate * (-1.0)
+                    self.publishStatus1StrAndValue(cmdType, "Driving", "Drive time in seconds of ", numParam1)
+
+                elif cmdType == self.cmdRotateLeft:
+                    # Rotate an angle. This will clear tracking till next command
```

Comment:
It would be easier to read if the code for each of these commands were in its own function.